### PR TITLE
Update version constraints to allow 4.0

### PR DESCRIPTION
**Pull request by miguelangelmorenochacon**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by miguelangelmorenochacon** [comment]
Any way to debug this? :)

File: modules/audit_config/versions.tf [hunk 1]
```diff
@@ -20,7 +20,7 @@ terraform {
 
     google = {
       source  = "hashicorp/google"
-      version = "~> 3.53"
+      version = ">= 3.53"
```
Comment (bharathkkb) [inline modules/audit_config/versions.tf:23]: Lets use `>= 3.53, < 5.0` here and throughout

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @miguelangelmorenochacon 
Lets also update the examples to use `~> 4.0`.

Build failure seems unrelated
```
$$$$$$ Validating the Terraform configuration files...
       
       Error: Could not load plugin
       
       
       Plugin reinitialization required. Please run "terraform init".
       
       Plugins are external binaries that Terraform uses to access and manipulate
       resources. The configuration provided requires plugins which can't be
       located,
       don't satisfy the version constraints, or are otherwise incompatible.
       
       Terraform automatically discovers provider requirements from your
       configuration, including providers used in child modules. To see the
       requirements and constraints, run "terraform providers".
       
       failed to instantiate provider "registry.terraform.io/hashicorp/random" to
       obtain schema: the cached package for registry.terraform.io/hashicorp/random
       3.1.0 (in .terraform/providers) does not match any of the checksums recorded
       in the dependency lock file
```

**Comment by miguelangelmorenochacon** [comment]
Looks like kitchen it's not able to upgrade to `~> 4.0`

```
+ kitchen destroy --test-base-path=test/integration
-----> Starting Test Kitchen (v2.7.2)
-----> Destroying <additive-local>...
$$$$$$ Reading the Terraform client version...
       Terraform v1.0.5
       on linux_amd64
       + provider registry.terraform.io/hashicorp/google v3.90.1
       + provider registry.terraform.io/hashicorp/google-beta v3.90.1
       + provider registry.terraform.io/hashicorp/random v3.1.0

       Your version of Terraform is out of date! The latest version
       is 1.0.11. You can update by downloading from https://www.terraform.io/downloads.html
$$$$$$ Finished reading the Terraform client version.
$$$$$$ Verifying the Terraform client version is in the supported interval of >= 0.11.4, < 1.1.0...
$$$$$$ Finished verifying the Terraform client version.
$$$$$$ Initializing the Terraform working directory...
       Initializing modules...

       Initializing the backend...

       Initializing provider plugins...
       - Reusing previous version of hashicorp/google from the dependency lock file
       - Reusing previous version of hashicorp/google-beta from the dependency lock file
       - Reusing previous version of hashicorp/random from the dependency lock file
       - Using previously-installed hashicorp/random v3.1.0
       ╷
       │ Error: Failed to query available provider packages
       │
       │ Could not retrieve the list of available versions for provider
       │ hashicorp/google: locked provider registry.terraform.io/hashicorp/google
       │ 3.90.1 does not match configured version constraint >= 3.53.0, ~> 4.0, <
       │ 5.0.0; must use terraform init -upgrade to allow selection of new versions
       ╵

       ╷
       │ Error: Failed to query available provider packages
       │
       │ Could not retrieve the list of available versions for provider
       │ hashicorp/google-beta: locked provider
       │ registry.terraform.io/hashicorp/google-beta 3.90.1 does not match
       │ configured version constraint ~> 4.0; must use terraform init -upgrade to
       │ allow selection of new versions

```

Not sure where I need to change this

**Comment by miguelangelmorenochacon** [comment]
Looks like we need to wait to this to be merged :)
- https://github.com/terraform-google-modules/terraform-google-project-factory/pull/637

**Review by bharathkkb — COMMENTED** [review]
@miguelangelmorenochacon since the project creation is part of setup we can continue to use 3.x and not block on the other release. I have updated this branch. Also looks like the [CLA is failing](https://github.com/terraform-google-modules/terraform-google-iam/pull/149/checks?check_run_id=4371064875), could you take a look?

**Comment by miguelangelmorenochacon** [comment]
> @miguelangelmorenochacon since the project creation is part of setup we can continue to use 3.x and not block on the other release. I have updated this branch. Also looks like the [CLA is failing](https://github.com/terraform-google-modules/terraform-google-iam/pull/149/checks?check_run_id=4371064875), could you take a look?

I've already signed the CLA aggrement

**Comment by bharathkkb** [comment]
@miguelangelmorenochacon Looks like it passed when I reran. Thanks!

**Review by bharathkkb — APPROVED** [review]

**Comment by upodroid** [comment]
Can you cut a new release? I'm currently blocked from using v4 of the provider. Thanks

**Pull request merged**